### PR TITLE
ISPN-14344 StoreMigrator support reading segmented SingleFileStores

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-14344

### CODE REVIEW

File: tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java [hunk 1]
```diff
@@ -13,19 +14,31 @@ class StoreIteratorFactory {
...
+      switch (type) {
          case JDBC_BINARY:
          case JDBC_MIXED:
          case JDBC_STRING:
             return new JdbcStoreReader(props);
+      }
+
+      Function<StoreProperties, StoreIterator> factory = fileStoreFactory(props.storeType());
+      if (props.isSegmented())
+         return new SegmentedFileStoreReader(props, factory);
```
Comment (jabolina) [inline tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java:27]: I am not completely aware, the store could have be segmented and of type RocksDb? If so we could have problems here. I mean, should we restrict to files?

**Review by jabolina — COMMENTED** [review]
Just a few comments.

File: tools/src/test/java/org/infinispan/tools/store/migrator/file/SingleFileStoreReaderTest.java [hunk 2]
```diff
@@ -22,9 +23,15 @@ public class SingleFileStoreReaderTest extends AbstractReaderTest {
    public Object[] factory() {
       return new Object[] {
             new SingleFileStoreReaderTest(),
-            new SingleFileStoreReaderTest().segmented(59),
             new SingleFileStoreReaderTest().majorVersion(9),
-            new SingleFileStoreReaderTest().majorVersion(9).segmented(59),
+            new SingleFileStoreReaderTest().majorVersion(9).targetSegments(59),
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply tools/src/test/java/org/infinispan/tools/store/migrator/file/SingleFileStoreReaderTest.java:27]: It doesn't really matter as the source is read into memory and then written to the target store via the Cache API, so having a discrepancy between the two doesn't make a difference.

File: tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java (hunk 1 above)
Comment (ryanemerson) [reply tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java:27]: It's not possible to use this approach with RocksDB stores as they use the RocksDB ColumnFamily as a basis of segmentation.

I did start to look into SIFS, but I ran into several issues and needed to get the PR up ASAP  for a customer.

**Review by jabolina — APPROVED** [review]

File: tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java (hunk 1 above)
Comment (wburns) [reply tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java:27]: We should probably throw an exception stating that RocksDB isn't supported with segmentation then.

**Review by wburns — COMMENTED** [review]
LGTM

**Review by wburns — APPROVED** [review]

File: tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java (hunk 1 above)
Comment (ryanemerson) [reply tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java]: Updated.

**Comment by wburns** [comment]
Integrated into main, thanks @ryanemerson !

**Pull request merged**